### PR TITLE
Add docs about testing automatic instrumentation

### PR DESCRIPTION
**Pull request by andrewlock**
## Summary of changes

Add docs about testing automatic instrumentation

## Reason for change

We didn't have any docs about it before and #4447 made some things easier

## Implementation details

Typing. I skipped the bits about running on macOS because I don't know how to do it 😄

### CODE REVIEW

File: docs/development/AutomaticInstrumentation.md [hunk 1]
```diff
@@ -276,3 +285,152 @@ There are some current limitations with what types/methods with our `CallTarget`
...
+
+# Build and run the integration tests, building only
+# your sample, and running only your new tests
+# You can choose whichever framework is appropriate
+./tracer/build.ps1 BuildAndRunWindowsIntegrationTests -buildConfiguration Debug -framework net462 -Filter MyNewIntegrationTests -SampleName Samples.MyNewSample
+```
+
+##### On macOs
+
+TBC
```
Comment (pierotibou) [inline docs/development/AutomaticInstrumentation.md:412]:
```suggestion
On MacOs, you won't be able to run all tests as some images aren't arm compatible (e.g. Kafka), some tests are meant to run on Windows only (using VM fixes that though). That said the process is quite the same as  for Windows:

```bash
# Build the tracer with your changes
# You need to run this every time you change your integration
# But not if you're just changing a test/sample
./tracer/build.sh BuildTracerHome -buildConfiguration Debug

# Build and run the integration tests, building only
# your sample, and running only your new tests
# You can choose whichever framework is appropriate
./tracer/build.sh BuildAndRunOSxIntegrationTests -buildConfiguration Debug -framework net6.0 -Filter MyNewIntegrationTests -SampleName Samples.MyNewSample
```
```

File: docs/development/AutomaticInstrumentation.md [hunk 2]
```diff
@@ -276,3 +285,152 @@ There are some current limitations with what types/methods with our `CallTarget`
...
+
+##### On Windows
+
+On Windows, we _don't_ typically run multi-api-version tests. There is experimental support locally for this, but you may run into issues. To run against a _single_ version (the default version lised in the sample) you can use:
+
+```bash
+# Build the tracer with your changes
+# You need to run this every time you change your integration
+# But not if you're just changing a test/sample
+./tracer/build.ps1 BuildTracerHome -buildConfiguration Debug
```
Comment (pierotibou) [inline docs/development/AutomaticInstrumentation.md:402]: TIL: `-buildConfiguration Debug`

File: docs/development/AutomaticInstrumentation.md [hunk 3]
```diff
@@ -276,3 +285,152 @@ There are some current limitations with what types/methods with our `CallTarget`
...
+# your sample, and running only your new tests
+# You can choose whichever framework is appropriate
+./tracer/build.ps1 BuildAndRunWindowsIntegrationTests -buildConfiguration Debug -framework net462 -Filter MyNewIntegrationTests -SampleName Samples.MyNewSample
+```
+
+##### On macOs
+
+TBC
+
+#### Testing in CI
```
Comment (pierotibou) [inline docs/development/AutomaticInstrumentation.md:414]: Should we remove line 44 and below that explains the same thing that you explain here?
`7. After testing locally, push to GitHub, and do a manual run in Azure Devops for your branch`

**Review by pierotibou — APPROVED** [review]

File: docs/development/AutomaticInstrumentation.md (hunk 2 above)
Comment (andrewlock) [reply docs/development/AutomaticInstrumentation.md:402]: Makes it easier if you need to attach a debugger 🙂

File: docs/development/AutomaticInstrumentation.md (hunk 3 above)
Comment (pierotibou) [reply docs/development/AutomaticInstrumentation.md:425]: LGTM, thanks again.

**Pull request merged**